src/java.base/share/classes/java/io/FileDescriptor.java

```diff
@@ -61,6 +60,11 @@ public final class FileDescriptor {
     private boolean closed;
 
     class Resource implements jdk.internal.crac.JDKResource {
+        private int originalFd = -1;
+        private String originalType;
+        private String originalPath;
+        private int originalFlags;
+        private long originalOffset;
```

Comment:
Regarding the design, path and offset are features of a file desriptor referring to the filesystem, but they do not make sense to socket FD.

Reply:
Yes, you could add special classes for Sockets vs. Files. It's easier, though, to treat this in single POJO and keep the path `null` if it does not make sense in this case.